### PR TITLE
Converting Rasa-Demo to 2.0 Format

### PR DESCRIPTION
**Pull request by b-quachtran**
To do:

- [x] Convert NLU files ([format guide](https://github.com/RasaHQ/rasa/blob/master/docs/docs/training-data-format.mdx#nlu-training-data))
- [x] Convert existing stories to new format ([format guide](https://github.com/RasaHQ/rasa/blob/master/docs/docs/rules.mdx#writing-a-rule))
- [x] Convert Forms to new format ([migration guide](https://github.com/RasaHQ/rasa/blob/master/docs/docs/migration-guide.mdx#form-policy))
- [x] Convert existing MappingPolicy to rules ([migration guide](https://github.com/RasaHQ/rasa/blob/master/docs/docs/migration-guide.mdx#migrating-from-the-mapping-policy))
- [x] Convert TwoStageFallbackPolicy to rules ([migration guide](https://github.com/RasaHQ/rasa/blob/master/docs/docs/migration-guide.mdx#migrating-from-the-two-stage-fallback-policy))
- [x] Migrate stories to rules where that makes sense
- [x] Update config

### CODE REVIEW

**Comment by b-quachtran** [comment]
@melindaloubser1 I've converted the NLU files to 2.0 (only thing missing is the lookup table file definition in the training data, since there doesn't seem to be support for that right now). 

I've put down a few to-do tasks, lmk which ones you can take up and if there's anything missing from that list.

**Comment by melindaloubser1** [comment]
Nice! The lookup tables are included inline in the yaml file now, no more txt files seperately. I can add those. I guess it makes sense to work in the order of the todos since they each depend on the previous to some degree. Would it be more helpful if I took an earlier one or a later one?

**Comment by b-quachtran** [comment]
Yah, can you take on the story conversion and test out the Core converter tool? I can work on the Forms migration in parallel since those changes will mostly be in the domain file.

**Comment by melindaloubser1** [comment]
sounds good!

**Comment by b-quachtran** [comment]
Data validation CI check is failing because of a bug with the new ResponseSelector format (https://github.com/RasaHQ/rasa/issues/6728)

**Comment by melindaloubser1** [comment]
@b-quachtran I'm seeing errors running `rasa data validate` - do you also?
A q about the test conversations: we really should have more test convos, but since we don't, how did you go about testing it all together?

**Comment by b-quachtran** [comment]
@melindaloubser1 Yeah I created an [issue](https://github.com/RasaHQ/rasa/issues/6728) for the `rasa data validate` bug. I've been testing locally through the shell but that's an action item for sure. I'll work on adding more test cases and update the ConveRT configs as well.

**Review by melindaloubser1 — CHANGES_REQUESTED** [review]
Few things aren't working:
-  Error when retrieving response selector intents:
 `Couldn't create message for response 'utter_out_of_scope'`
Is this part of the issue above?

- Evaluating all training stories shows we must have conflicts somewhere:
```
Evaluation Results on CONVERSATION level:
 Correct:          425 / 438
 F1-Score:         0.985
 Precision:        1.000
 Accuracy:         0.970
 In-data fraction: 0.927
Stories report saved to results/story_report.json.
Evaluation Results on ACTION level:
 Correct:          5883 / 5898
 F1-Score:         0.998
 Precision:        0.998
 Accuracy:         0.997
 In-data fraction: 0.927
```

General things:
-  we should use rules wherever it makes sense (i.e. for all the single-turn interactions). This especially since users will use this repo as an example of how to migrate a bot to 2.0, and we want to use our own best practices =)
- we should add test conversations for all our main skills. I know this is a lot on top of all the changes, but it will really be worth it when making changes in the new format, since we're more likely to make mistakes because we're less used to it.

**Comment by melindaloubser1** [comment]
We could potentially wait until sara is deployed in 2.0, then use rasa X to save test convos in the new format... but then we'd have to ship it without checking. I'll get started with a list of flows to add tests for so long.

**Comment by melindaloubser1** [comment]
Feel free to edit these lists

Tests to add (general categories):
- [ ] sales form
- [ ] getting started 
- [ ] newsletter form
- [ ] faqs (esp. the ones with slots)
- [ ] general questions (why_rasa, ask_how_contribute, etc.)
- [ ] handoff
- [ ] chitchat 
- [ ] out of scope
- [ ] fallback

Things that should probably have a rule associated with them (multiturn stories remain):
- sales form
- newsletter form
- ask_how_contribute
- ask_question_in_forum
- ask_when_next_event
- ask_which_events
- ask_why_contribute
- canthelp
- feedback
- greet
- technical_question
- thank
- why_rasa
- fallback
- source_code
- restart
- nlu_generation_tool_recommendation
- human_handoff

**Comment by b-quachtran** [comment]
@melindaloubser1 Training using Vova's branch https://github.com/RasaHQ/rasa/tree/fix-incomplete-rules works, so you can use that to test until the fix is packaged with the next 2.0 release

**Review by melindaloubser1 — COMMENTED** [review]
just leaving these comments for now

File: tests/test_stories.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#   - checkpoint: STORY_START
+#   - intent: technical_question
+#   - action: action_docs_search
+#   - slot_was_set:
+#     - docs_found: true
+#   - action: utter_ask_docs_help
+#   - intent: affirm
+#   - action: action_tag_docs_search
+#   - action: utter_great
+# - story: technical_question - no docs_found
```
[the remark this answers is not in the document]
Comment (b-quachtran) [reply tests/test_stories.yml:33]: Yep, this can be removed

File: tests/test_nlu.yml [hunk 2]
```diff
@@ -0,0 +1,793 @@
...
+nlu:
+- intent: affirm
+  examples: |
+    - yay
+    - yes accept please
+    - yeah
+    - oh cool
+    - yes please!
+    - Oh, ok
+    - of course
```
[the remark this answers is not in the document]
Comment (b-quachtran) [reply tests/test_nlu.yml:11]: I think we should keep the NLU tests, but I'll work on getting it up to date

File: data/rules/fallback.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+version: "2.0"
+rules:
+  - rule: Two stage fallback
+    steps:
+      - intent: nlu_fallback
+      - action: action_two_stage_fallback
+      - active_loop: action_two_stage_fallback
```
[the remark this answers is not in the document]
Comment (b-quachtran) [reply data/rules/fallback.yml:7]: It should do a revert by default after the correct intent is selected in the fallback, so I don't think so.

[the remark this answers is not in the document]
Comment (melindaloubser1) [reply data/rules/fallback.yml:7]: interesting. It doesn't need a `wait_for_user_input: false` either? Testing it atm

**Comment by b-quachtran** [comment]
@melindaloubser1 The rephrase issue with the Two Stage Fallback turned out to be a bug in the custom action code. It should be working now, although that open issue with `nlu_fallback` showing up as the predicted intent is still happening.

**Comment by melindaloubser1** [comment]
Look at that passing CI 🚀🚀

**Comment by melindaloubser1** [comment]
I am wondering why NLU results have decreased so much since last run here:  https://github.com/RasaHQ/rasa-demo/pull/552#issuecomment-652013024 - did we lose any data in this process?

**Comment by b-quachtran** [comment]
> I am wondering why NLU results have decreased so much since last run here: [#552 (comment)](https://github.com/RasaHQ/rasa-demo/pull/552#issuecomment-652013024) - did we lose any data in this process?

It looks like nlu_fallback is causing quite a bit of confusion for the intent cross validation, which is probably related to that open issue.

**Comment by melindaloubser1** [comment]
hmmm I think fallback_classifier needs to somehow get ignored during evaluation.

**Comment by b-quachtran** [comment]
> hmmm I think fallback_classifier needs to somehow get ignored during evaluation.

That seems like a bug, I'm following up [here](https://github.com/RasaHQ/rasa/issues/6285). 

Is there anything else that's causing a problem with Sara 2.0 in your testing?

**Comment by melindaloubser1** [comment]
Sorry just saw the q now - The many checkpoints in stories look like a potential converter bug, am trying to resolve them; also interruptions in forms weren't quite working, the rules for faqs specified `conversation_start: true` which I'm removing

**Comment by b-quachtran** [comment]
@melindaloubser1 I'm seeing a bunch of story contradictions being thrown during training, working through them at the moment but probably won't be resolved until Monday.

**Comment by b-quachtran** [comment]
@melindaloubser1 Context switching is working now and the rule conflicts are resolved. Can you review when you get a chance?

**Comment by melindaloubser1** [comment]
Will do. Have we opened an issue for the nlu_fallback confusion during evaluation anywhere?

**Comment by b-quachtran** [comment]
Yeah it's in the discussion in the issue Akela made [here](https://github.com/RasaHQ/rasa/issues/6285)

**Comment by koaning** [comment]
Just to make folks aware. I just tried running `rasa nlu test` locally and I'm seeing some strange bugs. 

```
> rasa test nlu --config config-templates/just-sparse.yml --cross-validation --runs 1 --folds 2 --out gridresults/just-sparse
...
...
...
Traceback (most recent call last):
  File "/home/vincent/Development/rasa-demo/venv/bin/rasa", line 8, in <module>
    sys.exit(main())
  File "/home/vincent/Development/rasa-demo/venv/lib/python3.7/site-packages/rasa/__main__.py", line 116, in main
    cmdline_arguments.func(cmdline_arguments)
  File "/home/vincent/Development/rasa-demo/venv/lib/python3.7/site-packages/rasa/cli/test.py", line 159, in run_nlu_test
    perform_nlu_cross_validation(config, nlu_data, output, vars(args))
  File "/home/vincent/Development/rasa-demo/venv/lib/python3.7/site-packages/rasa/test.py", line 276, in perform_nlu_cross_validation
    data, folds, nlu_config, output, **kwargs
  File "/home/vincent/Development/rasa-demo/venv/lib/python3.7/site-packages/rasa/nlu/test.py", line 1687, in cross_validate
    train,
  File "/home/vincent/Development/rasa-demo/venv/lib/python3.7/site-packages/rasa/nlu/test.py", line 1595, in combine_result
    ) = compute_metrics(interpreter, data)
  File "/home/vincent/Development/rasa-demo/venv/lib/python3.7/site-packages/rasa/nlu/test.py", line 1776, in compute_metrics
    interpreter, training_data
  File "/home/vincent/Development/rasa-demo/venv/lib/python3.7/site-packages/rasa/nlu/test.py", line 1310, in get_eval_data
    result = interpreter.parse(example.get(TEXT), only_output_properties=False)
  File "/home/vincent/Development/rasa-demo/venv/lib/python3.7/site-packages/rasa/nlu/model.py", line 398, in parse
    component.process(message, **self.context)
  File "/home/vincent/Development/rasa-demo/venv/lib/python3.7/site-packages/rasa/nlu/selectors/response_selector.py", line 382, in process
    self._resolve_intent_response_key(top_label) or top_label[INTENT_NAME_KEY]
  File "/home/vincent/Development/rasa-demo/venv/lib/python3.7/site-packages/rasa/nlu/selectors/response_selector.py", line 363, in _resolve_intent_response_key
    search_key = util.template_key_to_intent_response_key(key)
  File "/home/vincent/Development/rasa-demo/venv/lib/python3.7/site-packages/rasa/shared/nlu/training_data/util.py", line 145, in template_key_to_intent_response_key
    return template_key.split(UTTER_PREFIX)[1]
```

It's suggested [here](https://forum.rasa.com/t/rasa2-0-0rc2-no-configuration-file-was-provided-to-the-trainingdataimporter/35381) that it might be due to a data conversion error so I figured I'd check in.

**Comment by melindaloubser1** [comment]
@koaning I think this might be an environment thing, I'm not seeing that on 2.0.2

File: data/nlu/nlu.yml [hunk 4]
```diff
@@ -0,0 +1,2382 @@
...
+    - how can I leave a query in the forum?
+    - how can I post a question in the forum?
+    - how do I access the forum?
+    - i need information from posters in the forum
+    - ok send me to the forum
+    - to the forum
+    - need help on chatbot
+- intent: ask_which_events
+  examples: |
+    - Any other event like rasa meetup in future?
```
Comment (melindaloubser1) [inline data/nlu/nlu.yml:152]:
```suggestion
    - Any other event like rasa meetup in future?
    - Does Rasa host any events?
```

**Review by melindaloubser1 — APPROVED** [review]
besides the false positive story conflicts (`after ...` and `after action_listen` ) this looks good!

**Pull request merged**